### DISCUSSION TITLE
Equality of DefaultConnection and related objects

### DISCUSSION THREAD

**Discussion by pgr0ss** [graphql-java/graphql-java#2986] [Q&A] [answered]
We are currently migrating some code from using `SimpleListConnection` to our own logic for constructing `Connection` objects from more complicated database queries. It seems like we can still use the default classes (e.g. `DefaultEdge`, `DefaultPageInfo`, etc), rather than building our own implementations. However, one complication is that most of these objects don't seem to implement `equals`, which makes them more challenging to compare, especially in tests. Is there a reason for this?

For example:

- https://github.com/graphql-java/graphql-java/blob/8c947773944593c22f7f94b9aa6413801c3faebc/src/main/java/graphql/relay/DefaultConnection.java
- https://github.com/graphql-java/graphql-java/blob/8c947773944593c22f7f94b9aa6413801c3faebc/src/main/java/graphql/relay/DefaultEdge.java
- https://github.com/graphql-java/graphql-java/blob/8c947773944593c22f7f94b9aa6413801c3faebc/src/main/java/graphql/relay/DefaultPageInfo.java

Notably, `DefaultConnectionCursor` does implement it, though: https://github.com/graphql-java/graphql-java/blob/8c947773944593c22f7f94b9aa6413801c3faebc/src/main/java/graphql/relay/DefaultConnectionCursor.java#L24

Is this something I should open an issue for, or maybe even take a stab at contributing? Thanks!

**Comment by bbakerman** [comment, marked as answer]
>  However, one complication is that most of these objects don't seem to implement equals, which makes them more challenging to compare, especially in tests. Is there a reason for this?

The code in the Relay package is old and early and honestly not out best.  There is a no reason for not having .equals / .hashcode other than its not our best quality code.

if you could take a stab at contributing it would help us, you and others.  Open Source FTW!

**Reply by pgr0ss** [reply]
I opened a PR here: https://github.com/graphql-java/graphql-java/pull/2988
